src/tools/immutable_object_cache/Types.cc

```diff
@@ -90,6 +91,7 @@ void ObjectCacheReadData::encode_payload() {
   ceph::encode(read_len, payload);
   ceph::encode(pool_id, payload);
   ceph::encode(snap_id, payload);
+  ceph::encode(object_size, payload);
```

Comment:
Last time you comment: "You cannot inject new fields in the middle and maintain backwards compatibility. Plus, you also need to increment the version." If I put "ceph::encode(object_size, payload);" after "ceph::encode(pool_namespace, payload);", is it OK?

Reply:
Yes, it would need to go all the way at the end. You would still need to bump the encode version and in the decode you would only attempt to decode the object size field if the `version >= 2`.

Reply:
I put the encoded at the end,  and decode it by comparing verison(Static member variable) with the current version number(16.0.0-xxxxx) now.Please make sure that my writing is correct.

Reply:
Hi @dillaman,  I think I made a mistake. This version should come from the requester side version. However, after careful consideration, the parent cache runs on the local machine, and its communication is between  hook and daemon, both of which are on the same machine. Is it necessary to judge the version here? I don't think so. 
I am sorry that I am not familiar with these, and I will try to improve.

Reply:
> The versioning I am talking about is for the actual messages [1] not the Ceph build sha1.
> 
> [1] https://github.com/ceph/ceph/blob/master/src/tools/immutable_object_cache/Types.cc#L20
> [2] https://github.com/ceph/ceph/blob/master/src/tools/immutable_object_cache/Types.cc#L31

In the previous patch, mykola used the version of CEPH when registering  classes: ObjectCacheRegData()
(https://github.com/ceph/ceph/blob/0ef2a830b51127987829bf400a1b3d3e8418363f/src/tools/immutable_object_cache/Types.h#L60
https://github.com/ceph/ceph/blob/0ef2a830b51127987829bf400a1b3d3e8418363f/src/tools/immutable_object_cache/CacheClient.cc#L381
https://github.com/ceph/ceph/blob/0ef2a830b51127987829bf400a1b3d3e8418363f/src/tools/immutable_object_cache/CacheClient.cc#L383). 
Before, my code also converted the string of CEPH to int and then compared. The version of CEPH seems unreasonable. So should we redefine a version here, like version=1?

Reply:
That wasn't an on-the-wire format change -- it was an expectation change.